### PR TITLE
AI Assistant: Show upgrade banner when AI query endpoint responses with quota exceeded error

### PR DESCRIPTION
**Pull request by retrofox**
Fixes #
Part of https://github.com/Automattic/jetpack/issues/30842

## Proposed changes:

* AI Assistant: show upgrade banner when ai query endpoint responses with quota exceeded error

### Other information:

- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to '..'
*

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js [hunk 1]
```diff
@@ -19,10 +19,7 @@ const UpgradePrompt = () => {
 			buttonText={ 'Upgrade' }
 			checkoutUrl={ `${ window?.Jetpack_Editor_Initial_State?.adminUrl }admin.php?page=my-jetpack#/add-jetpack-ai` }
 			className={ 'jetpack-ai-upgrade-banner' }
-			description={ __(
-				'You have reached the limit of 20 free requests. Upgrade now to keep using Jetpack AI.',
-				'jetpack'
-			) }
+			description={ __( 'Upgrade now to keep using Jetpack AI.', 'jetpack' ) }
```
Comment (oskosk) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js:22]: Why this change?

**Review by oskosk — APPROVED** [review]

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js (hunk 1 above)
Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/upgrade-prompt.js:22]: the banner looks odd, to be honest. Let's try to get a final design for this iteration in a follow-up. 
image
